quran/models.py

```diff
@@ -1,19 +1,88 @@
 '''Models describing abstractions used at Quran app'''
-from django.db import migrations, models
+from os import path
+from django.db import models
+from django.template.defaultfilters import slugify
 
 
 class Surah(models.Model):
-    '''General information about Surahs'''
+    '''Model representing a surah from the Qur’an.'''
     number = models.PositiveIntegerField(primary_key=True)
     name = models.CharField(max_length=20)
     total_ayahs = models.PositiveIntegerField()
 
+    def __str__(self):
+        return "%s:(%s)" % (self.name, self.number)
+
 
 class Ayah(models.Model):
-    '''General information about Ayah'''
+    '''Model representing a ayah from the Qur’an.'''
     number = models.PositiveIntegerField()
     text = models.TextField()
     surah = models.ForeignKey(
         Surah,
         on_delete=models.CASCADE,
         related_name='ayahs')
+
+    def __str__(self):
+        return "%s: (%s:%s)" % (
+            self.surah.name, self.surah.number, self.number)
+
+
+class Reciter(models.Model):
+    '''Model representing a reciter'''
+    name = models.CharField(max_length=100)
+    quality = models.CharField(max_length=20, blank=True)
+    style = models.CharField(max_length=100, blank=True)
+    slug = models.SlugField()
+
+    def save(self, *args, **kwargs):
+        self.quality = self.quality.replace('/', 'p').replace(' ', '')
+        self.slug = slugify(self.name)
+        super(Reciter, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return self.name
+
+
+def audio_directory_path(instance, filename):
+    '''
+    the function generates path for
+    uploading audio file of ayahs to
+    MEDIA_ROOT/<reciter-slug>/<style>/<quality>/<surah-id>/<ayah-id>.mp3
+    '''
+    ayah = Ayah.objects.get(pk=instance.ayah_id)
```

Comment:
The code is pretty complicated, follow the instructions for `Recitation` model and update the code accordingly.